homeassistant/components/esphome/config_flow.py

```diff
@@ -78,7 +80,6 @@ async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
         self._host = entry.data[CONF_HOST]
         self._port = entry.data[CONF_PORT]
         self._password = entry.data[CONF_PASSWORD]
-        self._noise_psk = entry.data.get(CONF_NOISE_PSK)
```

Comment:
this is a re-auth. We know the key is wrong and will ask for a new one in `async_step_reauth_confirm`